### PR TITLE
test for nested directories and initial workdir requirement

### PR DESCRIPTION
**Pull request by rupertnash**
My first go at a conformance test, based on bug reported in #749 (Fixes https://github.com/common-workflow-language/common-workflow-language/issues/749 )

### CODE REVIEW

File: conformance_tests.yaml [hunk 1]
```diff
@@ -3399,3 +3399,24 @@
 - $import: tests/mixed-versions/test-index.yaml
 - $import: tests/loadContents/test-index.yaml
 - $import: tests/iwd/test-index.yaml
+
+- doc: >-
+    Test that nested directories, InitialWorkDirRequirement,
+    containers work correctly together. This is based on a reported
+    bug
+    https://github.com/common-workflow-language/common-workflow-language/issues/749
```
Comment (mr-c) [inline conformance_tests.yaml:3407]: This gets printed during test execution, so it really needs to be a single line.

Could you move the attribution into the CWL document itself? A YAML comment should be safe.

File: conformance_tests.yaml [hunk 2]
```diff
@@ -3399,3 +3399,24 @@
...
+      class: Directory
+      basename: my_dir
+      location: Any
+      listing:
+      - class: File
+        basename: my_file
+        location: Any
+        size: 0
+        checksum: sha1$da39a3ee5e6b4b0d3255bfef95601890afd80709
+  tags: [ initial_work_dir, subworkflow ]
```
Comment (mr-c) [inline conformance_tests.yaml:3422]: `tags` needs at least one of `command_line_tool`, `expression_tool` or `workflow` as per https://github.com/common-workflow-language/cwl-v1.2/blob/main/CONFORMANCE_TESTS.md#tags-for-conformance-tests

**Review by mr-c — CHANGES_REQUESTED** [review]
Thanks!

File: conformance_tests.yaml (hunk 1 above)
Comment (rupertnash) [reply conformance_tests.yaml]: Pedant in me says this is a single line, just a long one! Fixed as requested

**Comment by rupertnash** [comment]
Sorry - terrible commit message! Please squash when happy!

File: conformance_tests.yaml [hunk 3]
```diff
@@ -3399,3 +3399,20 @@
 - $import: tests/mixed-versions/test-index.yaml
 - $import: tests/loadContents/test-index.yaml
 - $import: tests/iwd/test-index.yaml
+
+- doc: Nested directories, InitialWorkDirRequirement, and containers work together
```
Comment (mr-c) [inline conformance_tests.yaml:3403]: >containers

Where? I don't see any `DockerRequirement`s. Might be good to add them, anyhow.

Comment (rupertnash) [reply conformance_tests.yaml:3403]: That is a reasonable point, but this was not in the origin issue. (See also our discussion about leaving in the `SubworkflowFeatureRequirement` even though it is not required).

Is there a preferred list of containers to use? This workflow only needs bash so almost anything will do.

Comment (mr-c) [reply conformance_tests.yaml:3403]: Yeah, but we should always have a container for consistency. Cloud runners will need one anyhow.

Yes, please use an existing container:
```
$ git grep dockerPull tests | awk '{print $NF}' | sed 's/"//g' | sort -u
alpine
bash:4.4.12
debian:10
debian:stable-slim
debian:stretch-slim
debian:stretch-slim,
python:2.7.15-alpine3.7
python:2-slim
python:2-slim
python:3
python:3-slim
```

**Comment by rupertnash** [comment]
I literally had just committed! Do we not need to add the hint to the CommandLineTool steps?

**Comment by mr-c** [comment]
> I literally had just committed!

Sorry!

> Do we not need to add the hint to the CommandLineTool steps?

We could have added them there instead, but I was feeling lazy and working from the web interface. Since we need to rebase or merge anyhow, we could move the hint down to them for simplicity.

**Review by mr-c — APPROVED** [review]

**Pull request merged**